core/src/main/java/org/mule/runtime/core/event/mutator/AbstractAddVariablePropertyProcessor.java

```diff
@@ -99,4 +104,11 @@ public void setValue(String value) {
     this.valueEvaluator = new AttributeEvaluator(value);
   }
 
+  public void setReturnDataType(DataType type) {
```

Comment:
Actually, this is defined for this specific element and its subchildren in `abstractAddPropertyProcessorType` in the xsd, plus there are tests testing this functionality outside of transformers.